### PR TITLE
Update string interpolation and fix some mistakes

### PR DESCRIPTION
**Pull request by Licenser**
Update our documentation to the new string interpolation syntax.

Also fix some minor mistakes.

### CODE REVIEW

File: docs/tremor-script/index.md [hunk 1]
```diff
@@ -116,9 +116,12 @@ The language does not support literal character or Unicode code-points at this t
 For strings tremor allows string interpolation, this means embedding code directly into strings to create strings out of them. This is a convenience as an alternative of using the string::format function.
```
Comment (mfelsche) [inline docs/tremor-script/index.md:116]: i think we need to either remove the mention of `string::format` or clarify the differences that we have now.

File: docs/workshop/examples/02_transform/README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ All other configuration is the same as per the passthrough example, and is elide
 
 ```trickle
 select {                                    # 1. We can inline new json-like document structures
-    "hello": "hi there {event.hello}",      # 2. Tremor supports flexible string interpolation useful for templating
+    "hello": "hi there ${event.hello}",      # 2. Tremor supports flexible string interpolation useful for templating
```
Comment (mfelsche) [inline docs/workshop/examples/02_transform/README.md:15]:
```suggestion
    "hello": "hi there #{event.hello}",      # 2. Tremor supports flexible string interpolation useful for templating
```

**Review by mfelsche — COMMENTED** [review]
i think we need some clarification about the differences with `string::format`, otherwise LGTM.

File: docs/tremor-script/index.md (hunk 1 above)
Comment (Licenser) [reply docs/tremor-script/index.md:116]: I think the sentence holds true it calls it out as an alternative but it also doesn't add much value so lets yank it and make sure no one is thrown of by it :+1:

File: docs/workshop/examples/02_transform/README.md (hunk 2 above)
Comment (Licenser) [reply docs/workshop/examples/02_transform/README.md:15]: :+1: good catch!

**Review by mfelsche — APPROVED** [review]

**Pull request merged**